jax/experimental/jax2tf/jax2tf.py

```diff
@@ -809,6 +810,79 @@ def _argminmax(fn, operand, axes, index_dtype):
 tf_impl[lax.cumsum_p] = tf.math.cumsum
 tf_impl[lax.cumprod_p] = tf.math.cumprod
 
+def _select_and_gather_add(tangents: TfVal,
+                           operand: TfVal,
+                           select_prim: core.Primitive,
+                           window_dimensions: Sequence[int],
+                           window_strides: Sequence[int],
+                           base_dilation: Sequence[int],
+                           window_dilation: Sequence[int],
+                           padding: Sequence[Tuple[int, int]],
+                           max_bits=64):
+  dtype = to_jax_dtype(operand.dtype)
+  nbits = dtypes.finfo(dtype).bits
+
+  assert nbits <= max_bits
+  double_word_reduction = nbits * 2 <= max_bits
+
+  const = lambda dtype, x: tf.constant(np.array(x), dtype)
+
+  if double_word_reduction:
+    word_dtype = lax.lax._UINT_DTYPES[nbits]
+    double_word_dtype = lax.lax._UINT_DTYPES[nbits * 2]
+
+    # Packs two values into a tuple.
+    def pack(a, b):
+      a = _bitcast_convert_type(a, word_dtype)
+      b = _bitcast_convert_type(b, word_dtype)
+      a = tfxla.convert_element_type(a, double_word_dtype)
+      b = tfxla.convert_element_type(b, double_word_dtype)
+      a = tf.bitwise.left_shift(a, const(double_word_dtype, nbits))
+      return tf.bitwise.bitwise_or(a, b)
+
+    # Unpacks the first element of a tuple.
+    def fst(t):
+      st = _shift_right_logical(t, const(double_word_dtype, nbits))
+      return _bitcast_convert_type(tfxla.convert_element_type(st, word_dtype), dtype)
+
+    # Unpacks the second element of a tuple.
+    def snd(t):
+      return _bitcast_convert_type(
+        tfxla.convert_element_type(t, word_dtype), dtype
+      )
+
+  else:
+    raise NotImplementedError("TODO: precision reduction case is not implemented")
+
+  def reducer(x, y):
+    which = tf_impl[select_prim]
+    return tf_impl[lax.select_p](which(fst(x), fst(y)), x=x, y=y)
```

Comment:
Why not use `tf.where` directly? I think it is less mysterious.

Reply:
My thinking here was that using `tf_impl[prim]` ensures that if we ever decide there is a better way to translate the primitive, the change is automatically applied to this as well. What do you think?
Otherwise, I don't mind changing this to be less cryptic.

Reply:
The translation of select_p and tf.where are both very simple. I'd suggest tf.where directly, but I do not feel strongly about it.